home tasks 1)tests for calculator 2)test day left by whois

3433825 opened the pull request.

3433825 committed:
home tasks 1)tests for calculator 2)test day left by whois

denyskulykov commented on days_left.py:
```diff
@@ -0,0 +1,14 @@
+    import whois
+
+    date_time_expire_obj = whois.whois(example_site).expiration_date
+    if type(date_time_expire_obj) is list:
```
isinstance()

denyskulykov commented on test_calc.py:
```diff
@@ -0,0 +1,37 @@
+import pytest
+from calculator import calc
+'''Test cases for calculator'''
```
'''Test cases for calculator
   positive and negative '''

3433825 committed:
next version 1)tests for calculator 2)test day left by whois

denyskulykov commented on calculator.py:
```diff
@@ -1,5 +1,43 @@
+        str = "".join(symbols_nearby)
+        all_neighbours.append(str)
+
+    if list(set(all_neighbours)&set(DIVIDING_BY_ZERO)):
```
Not need to convert set to list
empty set has the same logic like empty list for the bool operations

denyskulykov commented on check_reg_term.py:
```diff
@@ -0,0 +1,15 @@
+def check_example(example_site):
```
example_site is not a good name 
use just site

denyskulykov replied:
Why do we have this method?
what does it do?

denyskulykov commented on try_whois.py:
```diff
@@ -0,0 +1,17 @@
+def expect_whois(example_site):
+    import datetime
```
we should do import at the start of file

denyskulykov commented on test_positive_whois.py:
```diff
@@ -0,0 +1,12 @@
+import pytest
+from check_reg_term import check_example
+
+@pytest.mark.parametrize("example,expected", [("facebook.com", 3447),
```
need to change tests for date not for static days

denyskulykov commented on test_calc_positiv.py:
```diff
@@ -0,0 +1,31 @@
+import pytest
+from calculator import calc
+
+@pytest.mark.parametrize("example,expected", [("1 + 2", 3),
```
'example' is not a good name
use 'line'

denyskulykov commented on test_negative_whois.py:
```diff
@@ -0,0 +1,10 @@
+from check_reg_term import check_example
+
+
+@pytest.mark.parametrize("example,expected", [("facebook..com", "Wrong site name"),
```
the same for name
it is site or domain

denyskulykov replied:
Do we need it this function?

denyskulykov replied:
If you want to have some checkers in expect_whois, You can add them in expect_whois, not need to create new method

denyskulykov commented on try_whois.py:
```diff
@@ -0,0 +1,17 @@
+def expect_whois(example_site):
```
the same for name

3433825 committed:
fixing problems on whois library testing

3433825 committed:
fixing problems
test_calc_positiv.py - changed example to line
chec_reg_term.py - changed example_site to site
  - added blank row at the EOF
  - changed tests for date
  - combined all code for whois into one file

3433825 committed:
Implementation of the whois task

3433825 committed:
Implementation of the whois task

3433825 committed:
Implementation of the whois task

3433825 committed:
Implementation of the whois task
